### PR TITLE
Closes #1479 1479 consolidate metadata: implement `consolidate_metadata()`

### LINKED ISSUE #1479: Feature Request: Implement `consolidate_lookup_tables()`
### Feature Idea

Implement
```
consolidate_lookup_tables <- function(datasets,
                                      source_var = SOURCE,
                                      key_vars)
```
- `datasets` expects a named list of dataset, e.g., `datasets = list(company = global_adlb_params, study = adlb_params)`
- `key_vars` expects a list of variables, e.g., `key_vars = vars(LBTESTCD, LBSPEC, LBMETHOD)`

The function returns a dataset with one observation per key occurirng in any of the specified dataset. If a key occurs in more than one dataset, the observation from the last dataset is taken. The variable `SOURCE` should be added to the output dataset and set to the name of the dataset the observations is taken from.

Warnings should be issued if
- The keys are not a unique key of any of the datasets.
- The datasets contain different variables.
                                                           

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bundfussr** [comment]
The PR requires https://github.com/pharmaverse/admiraldev/issues/203.

File: R/consolidate_metadata.R [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+#' @details All observations of the input datasets are put together into a
+#'   single dataset. If a by group (defined by `key_vars`) exists in more than
+#'   one of the input datasets, the observation from the last dataset is
+#'   selected.
+#'
+#' @return A dataset which contains one row for each by group occuring in any of
+#'   the input datasets.
+#'
+#' @keywords create_aux
+#' @family create_aux
```
Comment (zdz2101) [inline R/consolidate_metadata.R:50]: Can be confusing if the word "metadata" is in the function name but not under the Metadata section of our reference page

File: R/consolidate_metadata.R [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+  }
+
+  data_order <- seq_len(length(datasets))
+  names(data_order) <- names(datasets)
+  all_data <- bind_rows(datasets, .id = as_label(source_var))
+  tmp_source_ord <- get_new_tmp_var(all_data, prefix = "tmp_source_ord")
+  all_data %>%
+    mutate(!!tmp_source_ord := data_order[!!source_var]) %>%
+    filter_extreme(
```
Comment (zdz2101) [inline R/consolidate_metadata.R:118]: Is there ever an instance where you might want to keep a non-last observation? For example in your code if HEIGHT was in both project and study but you wanted to keep project HEIGHT over the study HEIGHT? Having the ability to do the filter step after calling `consolidate_metadata()` seems more flexible but that might be me not entirely understanding the full picture of the use case

File: R/consolidate_metadata.R (hunk 1 above)
Comment (bundfussr) [reply R/consolidate_metadata.R:50]: Yes, but the "Metadata" section lists datasets and not functions. The description says "Auxiliary datasets providing definitions as input for derivations, e.g. grading criteria or dose frequencies". Thus `consolidate_metadata()` does not really fit into this section. I think it fits better into "Creating Auxiliary Datasets" although I must admit that it is not ideal.

File: R/consolidate_metadata.R (hunk 2 above)
Comment (bundfussr) [reply R/consolidate_metadata.R:118]: I am not aware of such a scenario. If you prefer project HEIGHT over study HEIGHT, why would you add HEIGHT to the study metadata at all?

File: R/consolidate_metadata.R [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+#'   of the dataset the observation is originating from.
+#'
+#'   *Permitted Values*: A symbol
+#'
+#' @param check_vars Check variables?
+#'
+#'   If `"message"`, `"warning"`, or `"error"` is specified, a message is issued
+#'   if the variable names differ across the input datasets (`datasets`).
+#'
+#'   *Permitted Values*: `"none"`, `"message"`, warning"`, `"error"`
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply R/consolidate_metadata.R]: fixed

**Review by zdz2101 — APPROVED** [review]

**Pull request merged**